modules/api.js

```diff
@@ -32,10 +32,39 @@ async function getMedicine(value) {
 // find best match
 async function getMedicineData(value) {
     const medicines = await fetch()
+    const rvgResults = stringifyJson(value)
+    const final = rvgResults.split(' ')
     const medicineNames = medicines.map(medicine => medicine.name)
     const medicine = stringSimilarity.findBestMatch(value, medicineNames).bestMatch
-    const medicineData = medicines.filter(meds => meds.name == medicine.target)
-    return medicineData
+    
+    if (rvgResults) {
+        const rvgData = medicines.filter(meds => {
+            if(meds.registrationNumber.includes(final[1])){
+               return meds
+            }
+        })
+        return rvgData
```

Comment:
is there a reason why you're returning twice?

Reply:
nope not needed

Reply:
misunderstood what the function did and it is needed, but it can be written a bit shorter. So it is resolved 😊